docs/madr/decisions/083-finalized-schemes-for-unified-envoy-resources-and-stats-naming-formats.md

```diff
@@ -0,0 +1,236 @@
+
+### Purpose
+
+Resources that directly map to user-facing Kuma resources.
+
+### Shape
+
+```
+kri_<resourceType>_<mesh>_<zone>_<namespace>_<resourceName>_<sectionName>
+```
+
+If a field is not present, keep its slot as an empty string so positions remain stable.
+
+### Allowed characters
+
+KRI uses a conservative charset suitable for URLs, xDS, and metrics. See "Constraints" in the KRI MADR.
+
+### Examples
+
+```
+kri_msvc_mesh-1_us-east-2_kuma-demo_backend_httpport
+kri_extsvc_mesh-1__kuma-system_es1_
+kri_mhttpr_mesh-1_us-east-2_kuma-demo_route-1_
+kri_zi__us-east-2_kuma-system_zi1_
+kri_ze__us-east-2_kuma-system_ze1_
+```
+
+These examples show user resources such as MeshService, MeshExternalService, MeshHTTPRoute, ZoneIngress, and ZoneEgress.
+
+### Notes
+
+- KRI avoids IPs and other high-cardinality bits.
+
+## Contextual format (proxy-local)
+
+### Purpose
+
+Some resources live only in the context of one proxy. The contextual format keeps names short and stable, and avoids mesh, zone, namespace, and proxy identity to prevent cardinality spikes. The chosen keyword is `self`.
+
+### Descriptor structure with the implementation adjustment
+
+We extend the descriptor with a **scope** token that marks the proxy kind:
+
+```
+self_<category>_<scope>_<rest>
+```
+
+- `<category>`: the feature area, for example `inbound` or `transparentproxy_passthrough`
+- `<scope>`: one of `dp` (sidecar dataplane), `zi` (ZoneIngress), `ze` (ZoneEgress)
+- `<rest>`: category-specific tail that may include `sectionName`, direction, IP version, etc.
+
+This adjustment replaces earlier examples like `self_inbound_httpport` with `self_inbound_dp_httpport`, so Zone can use the same category while staying unambiguous. *(Undocumented change made during implementation)*
+
+### Categories and shapes
+
+#### Inbounds
+
+```
+self_inbound_<scope>_<sectionName>
+```
+
+Examples:
+
+```
+self_inbound_dp_httpport
+self_inbound_dp_8080
+self_inbound_zi_10001
+self_inbound_ze_10001
+```
+
+The old `self_inbound_<sectionName>` form is now scoped. This keeps names short and avoids KRI for inbounds, which would raise metric cardinality.
+
+#### Transparent-proxy passthrough
+
+```
+self_transparentproxy_passthrough_<scope>_<direction>_ipv<ipVersion>
+```
+
+`<direction>` is `inbound` or `outbound`; `<ipVersion>` is `4` or `6`. Examples:
+
+```
+self_transparentproxy_passthrough_dp_inbound_ipv4
+self_transparentproxy_passthrough_dp_outbound_ipv6
+```
+
+Originally defined without a scope token; we append `<scope>` to align with the adjusted contextual scheme.
+
+### Why contextual here
+
+- Keeps names local to a proxy and avoids mesh/zone fields that are already metric labels
+- Prevents churn and excessive label sets in Prometheus and similar systems
+- Matches how policies refer to inbounds by `sectionName`
+
+### sectionName
+
+When present in contextual names, `sectionName` follows the same regex and rules as KRI. Use the port name if set, otherwise the port number.
```

Comment:
It would be good specifically for us (GUI) to be explicit about these being guaranteed to match the line format in the Envoy stats (that's the case right?)

Reply:
I'll adjust it in the followup PR